### PR TITLE
Trac3202

### PR DESCRIPTION
**Pull request by rpetry**
@olgamelnichuk, @alf239 - could you review please

### CODE REVIEW

File: atlas-loader/src/test/java/uk/ac/ebi/gxa/loader/handler/sdrf/TestAtlasLoadingDerivedArrayDataMatrixHandler.java [hunk 1]
```diff
@@ -110,7 +111,7 @@ public class TestAtlasLoadingDerivedArrayDataMatrixHandler extends TestCase {
         cache.setExperiment(new CreateExperimentStep().readExperiment(investigation, HashMultimap.<String, String>create()));
         final LoaderDAO dao = MockFactory.createLoaderDAO();
         new SourceStep().readSamples(investigation, cache, dao);
-        new AssayAndHybridizationStep().readAssays(investigation, cache, dao);
+        new AssayAndHybridizationStep().readAssays(investigation, cache, dao, (new AtlasMAGETABLoader()).isHTS(investigation));
```
Comment (alf239) [inline atlas-loader/src/test/java/uk/ac/ebi/gxa/loader/handler/sdrf/TestAtlasLoadingDerivedArrayDataMatrixHandler.java]: Can't we make it static? Looks like a dirty hack to me.

File: atlas-loader/src/test/java/uk/ac/ebi/gxa/loader/handler/sdrf/TestAtlasLoadingHybridizationHandler.java [hunk 2]
```diff
@@ -119,7 +120,7 @@ public class TestAtlasLoadingHybridizationHandler extends TestAssayHandler {
         cache.setExperiment(new CreateExperimentStep().readExperiment(investigation, HashMultimap.<String, String>create()));
         final LoaderDAO dao = MockFactory.createLoaderDAO();
         new SourceStep().readSamples(investigation, cache, dao);
-        new AssayAndHybridizationStep().readAssays(investigation, cache, dao);
+        new AssayAndHybridizationStep().readAssays(investigation, cache, dao, (new AtlasMAGETABLoader()).isHTS(investigation));
```
Comment (alf239) [inline atlas-loader/src/test/java/uk/ac/ebi/gxa/loader/handler/sdrf/TestAtlasLoadingHybridizationHandler.java]: Can't we make it static? Looks like a dirty hack to me.

**Comment by alf239** [comment]
Could we please get rid of the `(new AtlasMAGETABLoader()).` hack?

File: atlas-loader/src/test/java/uk/ac/ebi/gxa/loader/handler/sdrf/TestAtlasLoadingDerivedArrayDataMatrixHandler.java (hunk 1 above)
Comment (rpetry) [inline atlas-loader/src/test/java/uk/ac/ebi/gxa/loader/handler/sdrf/TestAtlasLoadingDerivedArrayDataMatrixHandler.java]: Done.

File: atlas-loader/src/test/java/uk/ac/ebi/gxa/loader/handler/sdrf/TestAtlasLoadingHybridizationHandler.java (hunk 2 above)
Comment (rpetry) [inline atlas-loader/src/test/java/uk/ac/ebi/gxa/loader/handler/sdrf/TestAtlasLoadingHybridizationHandler.java]: Done.

Comment (alf239) [inline atlas-loader/src/test/java/uk/ac/ebi/gxa/loader/handler/sdrf/TestAtlasLoadingHybridizationHandler.java]: Not really... Now done.

File: atlas-loader/src/test/java/uk/ac/ebi/gxa/loader/handler/sdrf/TestAtlasLoadingDerivedArrayDataMatrixHandler.java (hunk 1 above)
Comment (alf239) [inline atlas-loader/src/test/java/uk/ac/ebi/gxa/loader/handler/sdrf/TestAtlasLoadingDerivedArrayDataMatrixHandler.java]: Not really... Now done.

**Comment by alf239** [comment]
Cleaned up a bit; otherwise, happy with the change—feel free to merge in if you're okay with my changes (I know `AtlasLoadCache`,  `DataMatrixFileBuffer`, and `TestAtlasLoadingAssayHandler` are not in scope—but the change to them is dismal and hey, it's still 6 files against 7 you had!)

**Comment by alf239** [comment]
8 now, but still...

**Pull request merged**